pyo3-derive-backend/src/func.rs

```diff
@@ -77,7 +77,12 @@ pub(crate) fn impl_method_proto(
 ) -> TokenStream {
     let ret_ty = match &sig.output {
         syn::ReturnType::Default => quote! { () },
-        syn::ReturnType::Type(_, ty) => ty.to_token_stream(),
+        syn::ReturnType::Type(_, ty) => {
+            let mut ty = ty.clone();
+            // Insert lifetime to PyRef (i.e., PyRef<Self> -> PyRef<'p, Self>)
+            insert_lifetime_to_pyref(&mut ty);
```

Comment:
It would be nice if we could support adding lifetimes to references too. I notice that `get_arg_ty` does this.